File: Data Structures & Algorithms/validate-parentheses/submission-5.py
class Solution:
    def isValid(self, s: str) -> bool:
        partners = {")": "(", "}": "{", "]": "["}
        stack = []

        for char in s:
            if char in partners:
                if stack and stack[-1] == partners[char]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(char)

        return True if not stack else False            

File: Data Structures & Algorithms/daily-temperatures/submission-5.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<pair<int,int>> stk;
        vector<int> res(temperatures.size(),0);
        for (int i=0; i<temperatures.size(); i++) {
            while (!stk.empty() && temperatures[i] > stk.top().first) {
                auto temp = stk.top();
                stk.pop();
                res[temp.second] = i-temp.second;
            }
            stk.push({temperatures[i],i});
        }
        return res;
    }
};
  